dirlock/__init__.py

```diff
@@ -1,10 +1,75 @@
 import os
+import signal
 import time
+import atexit
 from datetime import datetime
+"""
+This module provides a directory-based locking mechanism.
+It allows for acquiring and releasing locks using directories,
+which can be useful in scenarios where file-based based
+operations can not be done atomically (e.g.
+distributed or cluster filesystem)
+
+The DirLock class provides methods to acquire and release locks,
+and it supports context management for easy usage.
+
+It also handles cleanup of locks on program exit or signal interrupts.
+"""
+
+
+# keep a list of currently acquired locks
+# so these can be cleaned up on exit
+_allActiveLocks = set()
+# we don't want to call the original handler
+original_sigint_handler = signal.getsignal(signal.SIGINT)
+original_sigterm_handler = signal.getsignal(signal.SIGTERM)
+
+
+# function to clean up all active locks
+def _clean_locks():
+    global _allActiveLocks
+    # Create a copy to avoid "Set changed size during iteration" error
+    locks_to_release = list(_allActiveLocks)
+    for dl in locks_to_release:
+        dl.release()
```

Comment:
I know this is just an internal function but I think it might be more intuitive for future development if the released locks are also removed from the list
```suggestion
        dl.release()
        _allActiveLocks.remove(dl)
```